### DISCUSSION TITLE
[教程] 如何让你的帖子更与众不同?

### DISCUSSION THREAD

**Discussion by Yersmagit** [share121/inter-knot#1436] [其他]
![封面](https://github.com/user-attachments/assets/72711900-88b8-4540-a884-375c8f2200db)

# 前言
**绳网 的帖子支持 Markdown ！** 
所以我们可以用一些简单的语法使你的贴子中含有不同格式的文本。
接下来，我将教你一些简单的 Markdown 使用方式。[^1]

> [!CAUTION]
> 开启浏览器的网页翻译功能可能导致本文显示出错，如果网页排版有任何异常请尝试**关闭网页翻译功能**！

> [!TIP]
> **文字教程看不懂的话可以展开看后面的示例哦！** 
> 关于使用 Markdown 的官方教程，请参阅[此处](https://docs.github.com/zh/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax)。

<img src="https://github.com/user-attachments/assets/98028886-3f8f-4eaf-bc45-893d8eb50e25" width="0.1" height="0.1">
<img src="https://github.com/user-attachments/assets/1e55c746-b248-49fc-8281-6e9c19f965df" width="0.1" height="0.1">
<img src="https://github.com/user-attachments/assets/363dcf90-53e1-4175-abb6-b742c8e88a4d" width="0.1" height="0.1">
<img src="https://github.com/user-attachments/assets/bbfd8895-3fcf-4f1b-a6af-548ddbfd2ecb" width="0.1" height="0.1">

# 教程
### 标题
- 你可以把井号 `#` 添加到你需要的标题前。
  井号数量可以在 1~6 之间，这决定了标题的层级结构与大小。

  <details>
  <summary> <b>点我展开查看示例</b> </summary>
  <br>
  示例：
    
  ```markdown
  # 我是一级标题
  ## 我是二级标题
  ### 我是三级标题
  ``` 
  效果如下：
  ![image](https://github.com/user-attachments/assets/98028886-3f8f-4eaf-bc45-893d8eb50e25)
  <br>
  </ul>
  </details>
  
> [!IMPORTANT]
> 井号 `#` 和文字间需要用一个空格隔开。
  
### 粗体和斜体
- 在文本前后各添加两个星号 `** **` 或两个下划线 `__ __` 可以使文本加粗，
  - 快捷键：`Command`+`B` (Mac) 或 `Ctrl`+`B` (Windows/Linux)；
- 在文本前后各添加一个星号 `* *` 或一个下划线 `_ _` 可以使文本倾斜，
  - 快捷键：`Command`+`I` (Mac) 或 `Ctrl`+`I` (Windows/Linux)。
  <br>
  <details>
  <summary> <b>点我展开查看示例</b> </summary>
  <br>
  示例：
  
  ```markdown 
  我是**粗体**文本
  我也是 __粗体__ 文本
  我是*斜体*文本
  我也是 _斜体_ 文本
  **我是又粗又*斜*的文本**
  __我也是又粗又 _斜_ 的文本__
  **我还是又粗又 _斜_ 的文本**
  ``` 
  效果如下：
  - 我是**粗体**文本
    我也是 __粗体__ 文本
    我是*斜体*文本
    我也是 _斜体_ 文本
    **我是又粗又*斜*的文本**
    __我也是又粗又 _斜_ 的文本__
    **我还是又粗又 _斜_ 的文本**

  </details>
  
> [!IMPORTANT]
> 如果使用了下划线加粗或倾斜文本，您必须在两边下划线的左右两侧各添加一个空格。
  
### 删除线
- 在文本前后各添加两个浪号 `~~ ~~` 可以给文本添加删除线。
  
  <details>
  <summary> <b>点我展开查看示例</b> </summary>
  <br>
  示例：
  
  ```markdown 
  我被~~删除~~了
  ``` 
  效果如下：
  - 我被~~删除~~了
  
  </details>
  
### 上标和下标
- 在文本前后依次添加 `<sup>` 和 `</sup>` 可以使此文本成为上标。
- 在文本前后依次添加 `<sub>` 和 `</sub>` 可以使此文本成为下标。
  
  <details>
  <summary> <b>点我展开查看示例</b> </summary>
  <br>
  示例：
  
  ```markdown
  我是<sup>上标</sup>
  我是<sub>下标</sub>
  ``` 
  效果如下：
  - 我是<sup>上标</sup>
    我是<sub>下标</sub>
  
  </details>
  
### 链接
- 你可以用这种方式添加链接：
  将需要添加链接的文本使用方括号 `[ ]` 括起来，再在其后方输入用括号 `( )` 括起来的链接。
  
  <details>
  <summary> <b>点我展开查看示例</b> </summary>
  <br>
  示例：
  
  ```markdown 
  点击[此处](https://github.com)进入 GitHub 官网
  ``` 
  效果如下：
  - 点击[此处](https://github.com)进入 GitHub 官网
   
  </details>

> [!IMPORTANT]
> 方括号和括号为英文半角符号！
  
### 图片或文件
- 单击编辑文本框右上角或底部的附加文件按钮可以添加图片或文件，或者直接拖放或粘贴以添加文件；
- 文件上传完成后，文件链接前的方括号 `[ ]` 内的内容为文件标题，可以按需修改。
> [!IMPORTANT]
> 文件上传需要一定时间，必须等待 `[Uploading]` 消失后再点击右下角的 `Start discussion` 或 `Comment` 按钮。
  
  
### 引用或告示
- 在一行文本的开头添加一个大于号和一个空格 `> ` 可以让这行文本变为引用的样式。
- 在引用内容的第一行输入 `[!NOTE]` 、 `[!TIP]` 、 `[!IMPORTANT]` 、 `[!WARNING]` 或 `[!CAUTION]` 可以使引用样式变为不同级别的告示。
  
  <details>
  <summary> <b>点我展开查看示例</b> </summary>
  <br>
  示例：
  
  ```markdown 
  > 这是一处引用。
  
  > [!NOTE]
  > 这是一处建议。
  
  > [!TIP]
  > 这是一处提示。
  
  > [!IMPORTANT]
  > 这是一处提醒。
  
  > [!WARNING]
  > 这是一处警示。
  
  > [!CAUTION]
  > 这是一处警告。
  ``` 
  效果如下：
  ![image](https://github.com/user-attachments/assets/1e55c746-b248-49fc-8281-6e9c19f965df)
  <br>
  </details>
  
> [!IMPORTANT]
> 输入时，两个引用或告示间必须用空行隔开。
  
### 表格
- 在文字间插入分隔符 `|` 可以绘制表格，
  在分隔符之间使用连接符 `-` 划分行标题。
  
  <details>
  <summary> <b>点我展开查看示例</b> </summary>
  <br>
  示例：
  
  ```markdown 
  
  |0|A|B|C|D|E|
  |-|-|-|-|-|-|
  |1|A1|B1|C1|D1|E1|
  |2|A2|B2|C2|D2|E2|
  |3|A3|B3|C3|D3|E3|
  ``` 
  效果如下：
  
  |0|A|B|C|D|E|
  |-|-|-|-|-|-|
  |1|A1|B1|C1|D1|E1|
  |2|A2|B2|C2|D2|E2|
  |3|A3|B3|C3|D3|E3|

  </details>

> [!TIP]
> 如果输出的表格格式有误，请尝试在表格开头添加一个空行。
  
### 代码
- 将单个反引号[^2]在文本前后添加可以标注句子中的代码或指令。
- 将三个反引号在文本的上下两行添加可以使代码或文本在整块代码框中显示；
  在第一个三反引号后输入代码的语言类型可以使用语法突出显示功能显示代码块。
    
  ``` 
  单反引号 `
  三反引号 ```
  ``` 
  
  <details>
  <summary> <b>点我展开查看示例</b> </summary>
  <br>
  示例：
  
  ````markdown 
  这里有一个 `/指令` 
  
  ```
    <script>
  document.write("我是一段HTML代码，我的开头没有html标签");
  </script>
  ```
  
  ```html
    <script>
  document.write("我是一段HTML代码，我的开头有html标签");
  </script>
  ```
  ````
  
  效果如下：
  - 这里有一个 `/指令` 
    
  - ```
      <script>
    document.write("我是一段HTML代码，我的开头没有html标签");
    </script>
    ```
    
  - ```html
      <script>
    document.write("我是一段HTML代码，我的开头有html标签");
    </script>
    ```

  </details>
  
> [!IMPORTANT]
> 输入时，第一个三反引号前需要有一个空行，否则可能导致格式错误。
  
#### 无序列表
- 在一行的开头输入一个连接符和一个空格 `- ` 可以创建一个一级列表；
- 在一行的开头直接使用空格对齐列表可以也可以快速创建列表。
  
  <details>
  <summary> <b>点我展开查看示例</b> </summary>
  <br>
  示例：
  
  ```markdown 
  - 我是一个一级列表
  - 我还是一个一级列表
    - 我是一个二级列表
      我还是一个二级列表，但我前面没有列表符
    - 我是一个二级列表，我前面有列表符
  ``` 
  效果如下：
  ![image](https://github.com/user-attachments/assets/363dcf90-53e1-4175-abb6-b742c8e88a4d)
  <br>
  </details>
  
> [!TIP]
> **看不懂上面的文字说明就看示例吧，注意每行行首空格的数量！** 
> 此处的连接符 `-` 也可以用加号 `+` 或星号 `*` 代替。

<br>

#### 有序列表
- 在每行开头输入数字、句点和空格可以快速创建有序列表。
- 多级列表的开头空格数量需要随上级列表开头字符数量做相应变化。

  <details>
  <summary> <b>点我展开查看示例</b> </summary>
  <br>
  示例：
  
  ```markdown 
  1. 一级列表
     - 二级列表
       二级列表
       - 三级列表
  2. 一级列表
  ``` 

  ```markdown 
  114. 我前面有八个字符（含空格）
       - 我的二级列表前需要有十个字符。
  ```

  效果如下：
  ![image](https://github.com/user-attachments/assets/bbfd8895-3fcf-4f1b-a6af-548ddbfd2ecb)
  <br>
  </details>
  
> [!IMPORTANT]
> 一级列表序号超过三位数字会使文本绘制在文本框以外而导致显示出错！

<br>

> [!NOTE]
> 实际使用前，建议先前往[ GitHub 个性化设置页面](https://github.com/settings/appearance)底部勾选<b> `Use a fixed-width (monospace) font when editing Markdown`（使用等宽字体编辑 Markdown 文本）选项</b>，这样可以让你编辑 Markdown 文本时不再眼花缭乱！
   
[^1]: <b>特别说明放在注脚，为的是不想让这个帖子变成像牛皮癣一样的存在。<br>这里简单说一下，这个贴是自制的，如果直接照搬官方教程怎么可能会有人反馈文案出错嘛！</b> 
[^2]: 双反引号输入快捷键： `Command`+`E`  (Mac) 或 `Ctrl`+`E` (Windows/Linux)，输入后可以自行按需调整数量。
[^3]: 此方式对图片、代码、引用等插入内容同样适用，**但对告示使用会导致文本格式变为引用。**

**Comment by lllliufeng** [comment]
我去，搬运[^1]
[^1]: Markdown is supported

**Reply by Yersmagit** [reply]
有参考，但自制哦

**Comment by Myitian** [comment]
> 用三个连续的连接符 `连接符---` 划分行标题。

不一定用3个`-`，可以是任意长度的
|1|2|3|4|5|6|
|-|--|---|----|-----|------|
|- 111111|-- 22222|--- 3333|---- 444|----- 55|------ 6|

**Reply by Yersmagit** [reply]
已修改！

> 用连接符 `-` 划分行标题。

**Comment by Lanxey** [comment]
_有没有类似BGM上的那种涂黑隐藏？_

**Reply by Yersmagit** [reply]
`▮▮■■■■■▬▬`
这种？

**Comment by lllliufeng** [comment]
没有吧

**Comment by Myitian** [comment]
【仅GitHub可用，绳网上只能显示代码】甚至可以有流程图（Mermaid）、地图（GeoJSON/TopoJSON）和 3D 模型（ASCII STL）
```mermaid
graph TD;
    A-->B;
    A-->C;
    B-->D;
    C-->D;
```
```geojson
{
  "type": "FeatureCollection",
  "features": [
    {
      "type": "Feature",
      "id": 1,
      "properties": {
        "ID": 0
      },
      "geometry": {
        "type": "Polygon",
        "coordinates": [
          [
              [-90,35],
              [-90,30],
              [-85,30],
              [-85,35],
              [-90,35]
          ]
        ]
      }
    }
  ]
}
```
```topojson
{
  "type": "Topology",
  "transform": {
    "scale": [0.0005000500050005, 0.00010001000100010001],
    "translate": [100, 0]
  },
  "objects": {
    "example": {
      "type": "GeometryCollection",
      "geometries": [
        {
          "type": "Point",
          "properties": {"prop0": "value0"},
          "coordinates": [4000, 5000]
        },
        {
          "type": "LineString",
          "properties": {"prop0": "value0", "prop1": 0},
          "arcs": [0]
        },
        {
          "type": "Polygon",
          "properties": {"prop0": "value0",
            "prop1": {"this": "that"}
          },
          "arcs": [[1]]
        }
      ]
    }
  },
  "arcs": [[[4000, 0], [1999, 9999], [2000, -9999], [2000, 9999]],[[0, 0], [0, 9999], [2000, 0], [0, -9999], [-2000, 0]]]
}
```
```stl
solid cube_corner
  facet normal 0.0 -1.0 0.0
    outer loop
      vertex 0.0 0.0 0.0
      vertex 1.0 0.0 0.0
      vertex 0.0 0.0 1.0
    endloop
  endfacet
  facet normal 0.0 0.0 -1.0
    outer loop
      vertex 0.0 0.0 0.0
      vertex 0.0 1.0 0.0
      vertex 1.0 0.0 0.0
    endloop
  endfacet
  facet normal -1.0 0.0 0.0
    outer loop
      vertex 0.0 0.0 0.0
      vertex 0.0 0.0 1.0
      vertex 0.0 1.0 0.0
    endloop
  endfacet
  facet normal 0.577 0.577 0.577
    outer loop
      vertex 1.0 0.0 0.0
      vertex 0.0 1.0 0.0
      vertex 0.0 0.0 1.0
    endloop
  endfacet
endsolid
```

**Reply by Buer-Nahida** [reply]
学到了，（还有个东西，@Myitian 艾特你的消息会强制推送到你的邮箱）
```mermaid
sequenceDiagram
Client->>Server: SYN
Server-->>Client: SYN,ACK
Client->>Server: ACK
```

**Comment by lllliufeng** [comment]
帖子中暂不支持啊（恼

**Reply by Myitian** [reply]
确实，仅GitHub可用，绳网上只能显示代码

**Comment by Buer-Nahida** [comment]
`插入代码、列表功能使用门槛较低，此处不做详细说明。`
# 门槛较低的教程在https://github.com/share121/inter-knot/discussions/1361

**Reply by Buer-Nahida** [reply]
（给我的帖子引下流）

**Comment by capacitor1** [comment]
> Reply::【仅GitHub可用，绳网上只能显示代码】甚至可以有流程图（Mermaid）、地图（GeoJSON/TopoJSON）和 3D 模型（ASCII STL）

GitHub还有64K字符限制。。我stl模型根本发不出来

**Comment by Myitian** [comment]
来点数学（
$P \left( \bigcup \limits_{i=1}^{+ \infty}A_{i}\right) = \prod \limits_{i=1}^{+ \infty}P{\left( A_{i}\right)} $
$\int \frac{1}{\sqrt{1-x^{2}}}\mathrm{d}x= \arcsin x +C $
$l \perp \beta ,l \subset \alpha \Rightarrow \alpha \perp \beta$
似乎又是仅GitHub可用，绳网上只能显示代码
详见[编写数学表达式 - GitHub 文档](https://docs.github.com/zh/get-started/writing-on-github/working-with-advanced-formatting/writing-mathematical-expressions)

**Comment by fairy-B** [comment]
# 完啦！

**Comment by silverwolf510trillion** [comment]
~~银狼~~

**Comment by Xmug** [comment]
好！

**Comment by thesilentaer** [comment]
### 蛙趣！

**Comment by haifanhen6** [comment]
### **没想到在绳网上也能学习**

**Comment by Orangered81** [comment]
# 谢谢喵<img src="https://upload-bbs.mihoyo.com/upload/2022/11/14/9e8bcae8c207bef5c02c176cdb6916ee_7651949051600440018.png" alt="阿姬低落" width="75" >

**Comment by aomengxin** [comment]
～～我是鲨臂～～

**Reply by Yersmagit** [reply]
哈哈用的应该是半角浪号`~~`，你输的是全角浪号`～～`
